src/main/java/hudson/plugins/promoted_builds/Status.java

```diff
@@ -106,19 +106,24 @@ public PromotionProcess getProcess() {
     public String getIcon(String size) {
         String baseName;
 
-        PromotionProcess p = getProcess();
-        if (p == null) {
-            // promotion process undefined (perhaps deleted?). fallback to the default icon
-            baseName = "star-gold";
-        } else {
-            Promotion l = getLast();
-            if (l!=null && l.getResult()!= Result.SUCCESS)
-                return Jenkins.RESOURCE_PATH+"/images/"+size+"/error.png";
-
-            baseName = p.getIcon();
-        }
-        return Jenkins.RESOURCE_PATH+"/plugin/promoted-builds/icons/"+size+"/"+ baseName +".png";
-    }
+		PromotionProcess p = getProcess();
+		if(p == null) {
+			// promotion process undefined (perhaps deleted?). fallback to the default icon
+			baseName = "star-gold";
+		}
+		else {
+			Promotion l = getLast();
+			if(l != null && l.getResult() != Result.SUCCESS)
+				return Jenkins.RESOURCE_PATH + "/images/" + size + "/error.png";
+
+			baseName = l.getIcon();
+			if(baseName == null) {
+				baseName = p.getIcon();
+
+			}
+		}
+		return baseName != null && !"none".equals(baseName) ? Jenkins.RESOURCE_PATH + "/plugin/promoted-builds/icons/" + size + "/" + baseName + ".png" : null;
```

Comment:
We don't want to remove the status as we still want to indicate that the de-promotion happened. We just want to remove the promotion badge-icon  when de-promotion happens. The concept of 'de-promotion' does not have to be implemented in the plugin, if the plugin is made flexible to allow the promotion-job to change the icon of the promotion badge.

I can make the method return "none" to indicate that no icon is assigned to prevent violation of contract. Would that be fine?